### PR TITLE
Wide & breaking changes - embeds/message detection

### PR DESCRIPTION
**Pull request by samtarling**
- Moved regex to own file
- Added utilities
- Added welcome/DM ident
- Added embeds
- Updated .env

This PR is likely to introduce breaking changes - **please** ensure you're locally up to date before doing a `checkout`, and review `.env.example` for changes

### CODE REVIEW

**Comment by samtarling** [comment]
@TamzinHadasa I changed a required check name (`Lint with mypy`) so this won't automatically pass - lemme know when you've reviewed it and I'll force it through 😄

File: utils.py [hunk 1]
```diff
@@ -128,3 +129,48 @@ async def safesend(ctx: Context,
...
                        + "utput too long to send as message. Sorry.]",
                        file=file)
+
+
+async def isDM(message: Message) -> bool:
+    """Helper function to see if this message was a DM"""
+    if not message.guild:
+        return True
+    else:
+        return False
```
Comment (TamzinHadasa) [inline utils.py:139]:
```suggestion
    return not message.guild
```

File: utils.py [hunk 2]
```diff
@@ -128,3 +129,48 @@ async def safesend(ctx: Context,
...
+
+
+async def isEmbed(message: Message) -> bool:
+    """Helper function to see if this message was an Embed"""
+    if message.embeds:
+        return True
+    else:
+        return False
```
Comment (TamzinHadasa) [inline utils.py:147]:
```suggestion
    return bool(message.embeds)
```
or `return not not message.embeds` (requires no global lookup) if you want to trade efficiency for readability. or just `return message.embeds` and tweak the typing,

File: utils.py [hunk 3]
```diff
@@ -128,3 +129,48 @@ async def safesend(ctx: Context,
...
+
+async def sendLoggerMessage(ctx: Context, content: str, error: bool) -> None:
+    """Helper function to send a DEBUG or ERROR message to the bot commands channel"""
+    # Only send if Dev = False
+    if not constants.DEV:
+        now_utc = await getUTC()
+        if error:
+            embed = Embed(
+                title="[ERROR MESSAGE]",
+                description=f"{content}",
```
Comment (TamzinHadasa) [inline utils.py:162]: Is the f-string here and at L170 necessary? Isn't `content` already a str?

**Review by TamzinHadasa — COMMENTED** [review]
All really cool stuff! I've noted a few TODOs, but do have one question and two suggestions. It's fine if you'd rather handle those suggestions and that question at a later date, but since they do marginally impact performance, thought I'd raise them now; happy to approve this if you'd rather I deal with them in a future commit.

File: bot.py [hunk 4]
```diff
@@ -20,41 +21,64 @@
...
-        return True
-    elif funfact_regex.match(message.content):
-        await message.reply("Is it *really* a fun fact though...?")
-        return True
-    elif devon_regex.match(message.content):
-        await message.reply("*That* had better have been an insult about Devon.")
-        return True
-    elif sock_regex.match(message.content):
-        await message.add_reaction("🧦")
+    if await utils.isDM(message):
```
[the remark this answers is not in the document]
Comment (samtarling) [reply bot.py:27]: not new behaviour :)

**Comment by samtarling** [comment]
Have resolved the required changes, and logged #29 so we can circle back :)

**Review by TamzinHadasa — APPROVED** [review]

**Pull request merged**